media/whip_server.go

```diff
@@ -209,6 +209,34 @@ func (s *WHIPServer) CreateWHIP(ctx context.Context, ssr *SwitchableSegmentReade
 		}
 		gatherDuration := time.Since(gatherStartTime)
 		clog.Infof(ctx, "Gathered %d tracks (%s) took=%v", len(trackCodecs), strings.Join(trackCodecs, ", "), gatherDuration)
+
+		go func() {
+			// Periodically check whip stats and write logs and metrics
+			for !mediaState.IsClosed() {
+				time.Sleep(time.Second * 5)
+				statsReport := peerConnection.GetStats()
+
+				for _, stat := range statsReport {
+					switch s := stat.(type) {
+					case webrtc.TransportStats:
+						if monitor.Enabled {
+							monitor.AIWhipTransportBytesReceived(streamName, int64(s.BytesReceived))
+							monitor.AIWhipTransportBytesSent(streamName, int64(s.BytesSent))
+							monitor.AIWhipTransportPacketsReceived(streamName, int64(s.PacketsReceived))
+							monitor.AIWhipTransportPacketsSent(streamName, int64(s.PacketsSent))
+						}
+						clog.Info(ctx, "whip TransportStats", "ID", s.ID, "bytes_received", s.BytesReceived, "bytes_sent", s.BytesSent, "packets_received", s.PacketsReceived, "packets_sent", s.PacketsSent, "dtls_state", s.DTLSState, "ice_state", s.ICEState)
+					// not seeing these showing up currently, but hopefully we can fix whatever is causing that
```

Comment:
@emranemran @j0sh any idea why we don't see these `InboundRTPStreamStats` stats coming through? I went back and forth with gpt ha and it seemed like if we're receiving RTP then we should get these. And we surely must be receiving RTP if the stream is healthy and we're segmenting etc.